src/Spanner/SpannerClient.php

```diff
@@ -0,0 +1,497 @@
+     *     'startType' => KeyRange::TYPE_OPEN,
+     *     'start' => ['Bob'],
+     *     'endType' => KeyRange::TYPE_OPEN,
+     *     'end' => ['Jill']
+     * ]);
+     * ```
+     *
+     * @param array $options [optional] {
+     *     Configuration Options.
+     *
+     *     @type string $startType Either "open" or "closed". Use constants
+     *           `KeyRange::TYPE_OPEN` and `KeyRange::TYPE_CLOSED` for
+     *           guaranteed correctness.
+     *     @type array $start The key with which to start the range.
+     *     @type string $endType Either "open" or "closed". Use constants
+     *           `KeyRange::TYPE_OPEN` and `KeyRange::TYPE_CLOSED` for
+     *           guaranteed correctness.
+     *     @type array $end The key with which to end the range.
+     * }
+     * @return KeyRange
+     */
+    public function keyRange(array $options = [])
+    {
+        return new KeyRange($options);
+    }
+
+    /**
+     * Create a Bytes object.
+     *
+     * Example:
+     * ```
+     * $bytes = $spanner->bytes('hello world');
+     * ```
+     *
+     * @param string|resource|StreamInterface $value The bytes value.
+     * @return Bytes
+     */
+    public function bytes($bytes)
+    {
+        return new Bytes($bytes);
+    }
+
+    /**
+     * Create a Date object.
+     *
+     * Example:
+     * ```
+     * $date = $spanner->date(new \DateTime('1995-02-04'));
+     * ```
+     *
+     * @param \DateTimeInterface $value The date value.
+     * @return Date
+     */
+    public function date(\DateTimeInterface $date)
+    {
+        return new Date($date);
+    }
+
+    /**
+     * Create a Timestamp object.
+     *
+     * Example:
+     * ```
+     * $timestamp = $spanner->timestamp(new \DateTime('2003-02-05 11:15:02.421827Z'));
+     * ```
+     *
+     * @param \DateTimeInterface $value The timestamp value.
+     * @param int $nanoSeconds [optional] The number of nanoseconds in the timestamp.
+     * @return Timestamp
+     */
+    public function timestamp(\DateTimeInterface $timestamp, $nanoSeconds = null)
+    {
+        return new Timestamp($timestamp, $nanoSeconds);
+    }
+
+    /**
+     * Create an Int64 object. This can be used to work with 64 bit integers as
+     * a string value while on a 32 bit platform.
+     *
+     * Example:
+     * ```
+     * $int64 = $spanner->int64('9223372036854775807');
+     * ```
+     *
+     * @param string $value
+     * @return Int64
+     */
+    public function int64($value)
+    {
+        return new Int64($value);
+    }
+
+    /**
+     * Create a Duration object.
+     *
+     * Example:
+     * ```
+     * $duration = $spanner->duration(100, 00001);
+     * ```
+     *
+     * @param int $seconds The number of seconds in the duration.
+     * @param int $nanos [optional] The number of nanoseconds in the duration.
+     *        **Defaults to** `0`.
+     * @return Duration
+     */
+    public function duration($seconds, $nanos = 0)
```

Comment:
It is a wrapper for the [protobuf Duration](https://github.com/google/protobuf/blob/master/src/google/protobuf/duration.proto) type, used in [TransactionOptions#readOnly](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.TransactionOptions.ReadOnly).